classes/class.ilMumieTaskGradeListGUI.php

```diff
@@ -0,0 +1,207 @@
+                        $this->tpl->setVariable("VAL_DATE", substr($xGrade->timestamp, 0, 10) . " - " . substr($xGrade->timestamp, 11, 5));
+                        $this->tpl->setCurrentBlock("tbl_content");
+                        $this->tpl->parseCurrentBlock();
+                    }
+                }
+            }
+        }
+
+        $this->enable('header');
+        $this->enable('sort');
+        $this->setEnableHeader(true);
+    }
+
+    public static function overrideGrade($parentObj)
+    {
+        global $ilDB, $DIC, $lng;
+        $percentage = $_GET['newGrade'];
+        $DIC->database()->update(
+            'ut_lp_marks',
+            array(
+                "status_changed" => array('text', date("Y-m-d H:i:s", $_GET['timestamp'])),
+                "mark" => array('int', $percentage),
+            ),
+            array(
+                'obj_id' => array('int', $parentObj->object->getId()),
+                'usr_id' => array('int', $_GET['user_id']),
+            )
+        );
+        require_once('./Customizing/global/plugins/Services/Repository/RepositoryObject/MumieTask/classes/class.ilMumieTaskGradeSync.php');
+        require_once('Customizing/global/plugins/Services/Repository/RepositoryObject/MumieTask/classes/class.ilMumieTaskIdHashingService.php');
+        $hashed_user = ilMumieTaskIdHashingService::getHashForUser($_GET["user_id"], $parentObj->object);
+        $gradesync  = new ilMumieTaskGradeSync($parentObj->object, false);
+        if (!$gradesync->wasGradeOverriden($_GET["user_id"])) {
+            $ilDB->insert(
+                "xmum_grade_override",
+                array(
+                    'task_id' => array('integer', $parentObj->object->getId()),
+                    'usr_id' => array('text', $hashed_user),
+                    'new_grade' => array('integer', $percentage)
+                )
+            );
+        } else {
+            $ilDB->update(
+                "xmum_grade_override",
+                array(
+                    'new_grade' => array('integer', $percentage)
+                ),
+                array(
+                    'task_id' => array('integer', $parentObj->object->getId()),
+                    'usr_id' => array('text', $hashed_user),
+                )
+            );
+        }
+        $result = $ilDB->query("SELECT firstname, lastname FROM usr_data WHERE usr_id = ". $ilDB->quote($_GET['user_id'], "integer"));
+        $names = $ilDB->fetchAssoc($result);
+        ilUtil::sendSuccess($lng->txt('rep_robj_xmum_frm_list_successfull_update') . " " . $names["firstname"] . ",  " . $names["lastname"] . " " .  $lng->txt('rep_robj_xmum_frm_list_to') . " " . $percentage);
+    }
+
+    public function insert($a_tpl)
+    {
+        $a_tpl->setCurrentBlock("prop_custom");
+        $a_tpl->setVariable("CUSTOM_CONTENT", $this->render());
+        $a_tpl->parseCurrentBlock();
+    }
```

Comment:
where is this used?

Reply:
Can you tell me which file and which line? I can't find it myself in the code and my IDE cannot find any usage either

Reply:
it is called in the table functions, it is buried somewhere in ilias code for getHTML i think but if i remove it the list breaks